src/configuration.ts

```diff
@@ -41,4 +43,76 @@ export const getConfiguration = () => {
 	};
 };
 
+export class DefaultConfigurationClass {
+	#messageBus: MessageBus;
+	constructor(messageBus: MessageBus) {
+		this.#messageBus = messageBus;
+		this.#messageBus.subscribe(MessageKind.extensionActivated, async () => {
+			const repositoryPathFromConfig = getConfiguration().repositoryPath;
+			if (!repositoryPathFromConfig) {
+				try {
+					const repositoryPath = await Promise.race([
+						this.getGitRemote(),
+						this.timeout(4000),
+					]);
+
+					if (repositoryPath) {
+						await vscode.workspace
+							.getConfiguration('intuita')
+							.update(
+								'repositoryPath',
+								repositoryPath,
+								vscode.ConfigurationTarget.Global,
+							);
+					}
+				} catch (error) {
+					if (error instanceof Error) {
+						vscode.window.showErrorMessage(error.message);
+					}
+					console.error('error', error);
+				}
+			}
+		});
+	}
+	getGitRemote = () =>
+		// eslint-disable-next-line no-async-promise-executor
+		new Promise(async (resolve, reject) => {
```

Comment:
actually I think it's a serious antipattern: https://stackoverflow.com/questions/23803743/what-is-the-explicit-promise-construction-antipattern-and-how-do-i-avoid-it/23803744#23803744

I think you can make it an async function and create the promise within it when needed.